File: data/src/main/java/dev/olog/data/api/lastfm/artist/ArtistInfo.java

package dev.olog.data.api.lastfm.artist;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class ArtistInfo {

    public Artist artist;

    public static class Artist {

        public String name;

        public String mbid;

        public String url;

        public List<Image> image = new ArrayList<>();

        public Bio bio;

        public static class Image {

            @SerializedName("#text")
            public String text;
            public String size;
        }

        public static class Bio {

            public Links links;
            public String published;
            public String summary;
            public String content;

            public static class Links {

                public Link link;

                public static class Link {

                    @SerializedName("#text")
                    public String text;
                    public String rel;
                    public String href;

                }

            }


        }

    }


}

File: data/src/main/java/dev/olog/data/api/lastfm/artist/ArtistInfo.kt
package dev.olog.data.api.lastfm.artist

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class ArtistInfo(
    val artist: Artist? = null,
) {

    @Serializable
    class Artist(
        val name: String? = null,
        val mbid: String? = null,
        val url: String? = null,
        val image: List<Image>? = null,
        val bio: Bio? = null,
    )

    @Serializable
    class Image(
        @SerialName("#text")
        val text: String? = null,
        val size: String? = null,
    )

    @Serializable
    class Bio(
        val links: Links? = null,
        val published: String? = null,
        val summary: String? = null,
        val content: String? = null,
    )

    @Serializable
    class Links(
        val link: Link? = null,
    )

    @Serializable
    class Link(
        @SerialName("#text")
        val text: String? = null,
        val rel: String? = null,
        val href: String? = null,
    )

}